### PR TITLE
Text input related design browser fixes

### PR DESCRIPTION
**Pull request by luxaritas**

### CODE REVIEW

File: src/eterna/mode/DesignBrowser/DataCol.ts [hunk 1]
```diff
@@ -19,23 +19,31 @@ import {SortOrder} from './SortOptions';
...
 export default class DataCol extends ContainerObject {
     public readonly sortOrderChanged = new Signal<SortOrder>();
     public readonly filtersChanged = new UnitSignal();
     public readonly voteChanged = new Signal<number>();
     public readonly category: DesignCategory;
 
-    constructor(dataType: DesignBrowserDataType, category: DesignCategory,
-        dataWidth: number, fonttype: string,
-        fontSize: number, sortable: boolean) {
+    constructor(props: DataColProps) {
```
Comment (everyday847) [inline src/eterna/mode/DesignBrowser/DataCol.ts:37]: So the real non-cosmetic change throughout here is essentially the addition of the `domParent` -- can you explain why DataCol needs to know that? Just so the TextInputObjects it creates share its parent?

File: src/eterna/mode/DesignBrowser/DesignBrowserMode.ts [hunk 2]
```diff
@@ -872,9 +872,20 @@ export default class DesignBrowserMode extends GameMode {
                 }
 
                 let column: DataCol;
+                const baseParams = {
```
Comment (everyday847) [inline src/eterna/mode/DesignBrowser/DesignBrowserMode.ts:875]: nice addition

File: src/eterna/ui/TextInputObject.ts [hunk 3]
```diff
@@ -56,7 +58,11 @@ export default class TextInputObject extends DOMObject<HTMLInputElement | HTMLTe
                 setTimeout(() => {
                     this.destroyFakeTextInput();
                     this._obj.style.visibility = 'visible';
-                    this._obj.focus();
+                    // On Chrome, it seems there's a bug (?) that causes the input to zoom into focus
```
Comment (everyday847) [inline src/eterna/ui/TextInputObject.ts:61]: bizarre!

**Review by everyday847 — APPROVED** [review]
Looks fine; thanks for the bug fixes

File: src/eterna/mode/DesignBrowser/DataCol.ts (hunk 1 above)
Comment (luxaritas) [reply src/eterna/mode/DesignBrowser/DataCol.ts:37]: Yeah it's so we can pass it to TextInputObject, which passes it to DOMObject, which uses it to insert itself into the DOM at the correct point (specifically, as a child of the masked div created by ScrollContainer)

**Pull request merged**